pybop/builders/pybamm.py

```diff
@@ -35,9 +37,21 @@ def set_simulation(
         self._solver = solver or model.default_solver
 
     def add_cost(self, cost: PybammCost, weight: float = 1.0) -> None:
+        """
+        Add a cost to the problem with optional weighting.
+        """
         self._costs.append(cost)
         self._cost_weights.append(weight)
 
+    def add_posterior(self, cost: PybammCost, weight: float = 1.0) -> None:
```

Comment:
I'm not a fan of having a separate function for this, users could easily just pass a likelihood to `add_cost` with the same effect? If a prior is provided then you add this to make a posterior, if a prior is not provided then you just treat the likelihood as a cost. What do you think?

Reply:
yea, sounds like a good idea. Feel free to merge this in whenever you are ready